frontend/src/pages/VendorsByMarketPage.js

```diff
@@ -23,9 +26,39 @@ const VendorsByMarketPage = () => {
     fetchMarket(marketId);
   }, [marketId]);
 
+  const filteredVendorArray =
+    searchTerm.length === 0
+      ? market.vendors
+      : market.vendors.filter(
+          (vendor) =>
+            vendor.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
+            vendor.products
+              .join(" ")
+              .toLowerCase()
+              .includes(searchTerm.toLowerCase()) ||
+            vendor.intro.toLowerCase().includes(searchTerm.toLowerCase()) ||
+            vendor.introductionLong
+              .toLowerCase()
+              .includes(searchTerm.toLowerCase())
+        );
+
   const renderVendorList = () => {
     if (market.vendors && market.vendors.length > 0) {
-      return <VendorListOfOneMarket market={market} />;
+      return (
+        <>
+          <div className="search-area">
```

Comment:
Ez a keresés rész is általánosnak tűnik, nem lehet valahogy componentbe kitenni? és csak felhasználni ott ahol szeretnénk keresni. Nyilván általánosítani kell hogy mi alapján akarunk keresgélni